language-support/java/codegen/src/ledger-tests/scala/com/digitalasset/CodegenLedgerTest.scala

```diff
@@ -164,4 +167,26 @@ class CodegenLedgerTest extends AsyncFlatSpec with Matchers with TestResourceCon
     wolpertinger.observers shouldBe empty
   }
 
+  it should "be able to create multi-party templates" in withClient { client =>
+    val multi = new MultiParty(Alice, Bob)
+    sendCmd(client, list(Alice, Bob), list[String](), multi.create());
+
+    val read :: _ = readActiveContracts(MultiParty.Contract.fromCreatedEvent)(client)
```

Comment:
Any reason to use a pattern here? The error is probably easier to read using `.head`:

```
scala> val read :: _ = List.empty[Int]
scala.MatchError: List() (of class scala.collection.immutable.Nil$)
  ... 28 elided

scala> val read = List.empty[Int].head
java.util.NoSuchElementException: head of empty list
  at scala.collection.immutable.Nil$.head(List.scala:469)
  at scala.collection.immutable.Nil$.head(List.scala:466)
  ... 28 elided
```

This also protects from some head-scratching situation if the concrete return type of `readActiveContracts` changes:

```
scala> val read :: _ = Vector(1, 2, 3).asInstanceOf[Seq[Int]]
scala.MatchError: Vector(1, 2, 3) (of class scala.collection.immutable.Vector)
  ... 28 elided

scala> val read = Vector(1, 2, 3).asInstanceOf[Seq[Int]].head
read: Int = 1
```

Reply:
Interesting, iirc in Haskell it’s the opposite. The pattern gives you a better error message than calling `head`.

Reply:
Changed.